Sources/Sentry/SentryTracer.m

```diff
@@ -133,6 +138,27 @@ - (void)captureTransaction
     }];
 }
 
+- (void)addMeasurements:(SentryTransaction *)transaction
+{
+    if (SentrySDK.appStartMeasurement != nil) {
+        NSString *appStartMeasurementType = nil;
+
+        if ([SentrySDK.appStartMeasurement.type isEqualToString:SentryAppStartTypeCold]) {
+            appStartMeasurementType = @"app_start_time_cold";
+        } else if ([SentrySDK.appStartMeasurement.type isEqualToString:SentryAppStartTypeWarm]) {
+            appStartMeasurementType = @"app_start_time_warm";
+        }
+
+        if (appStartMeasurementType != nil) {
+            [transaction
+                setMeasurementValue:@{ @"value" : @(SentrySDK.appStartMeasurement.duration * 1000) }
```

Comment:
This needs to be synchronized.

While we're here (after the null check above) another transaction could be captured and already set `appStartMeasurement=nil`.

If we want to guarantee that `appStartMeasurement` is used once, I think an atomic reference swap is the best approach where one thread will "take over" the value and run with it to add to its transaction.

Reply:
Good catch. I fixed it with double-checked locking and swapping out the reference.